modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/NestedLoopJoinNode.java

```diff
@@ -462,7 +463,9 @@ private RightJoin(
         @Override
         protected void rewindInternal() {
             left = null;
-            rightNotMatchedIndexes.clear();
+            if (rightNotMatchedIndexes != null) {
```

Comment:
I don't see any related changes, which can lead to the change sequentially. Which test show the problem?

Reply:
one of the test from SqlLogic suite. I've got NPE here and decided, that the problem was in missed check. Turns out the issue was a bit trickier... I changed execution test in order to cover this case